### PR TITLE
Adding Test Coverage Report, Static Code Analysis & Code Style Check #14

### PR DESCRIPTION
**Pull request by pliakas**
Adding JaCoCo Report for Core Modules & publish badges to 'Readme.md'

 @anantk24 You can use this configuration for JaCoCo.

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -28,3 +28,40 @@ jobs:
...
+      with:
+        generate-branches-badge: true
+        jacoco-csv-file: core/vertx/target/site/jacoco/jacoco.csv
+        coverage-badge-filename: jacoco3.svg
+        branches-badge-filename: branches3.svg
+
+    - name: Commit the badge (if it changed)
+      run: |
+        if [[ `git status --porcelain` ]]; then
+          git config --global user.name 'Thomas Pliakas'
```
Comment (karianna) [inline .github/workflows/maven.yml:62]: We probably need to use an account that's not your personal one here :-).  Is there another way?

File: pom.xml [hunk 2]
```diff
@@ -75,6 +75,12 @@
         <maven.compiler.release>11</maven.compiler.release>
         <skipUnpack>false</skipUnpack>
         <gctoolkit-testdata-version>1.0.2</gctoolkit-testdata-version>
+<!--        <gctoolkit-testdata-version>1.0.3-SNAPSHOT</gctoolkit-testdata-version>-->
```
Comment (karianna) [inline pom.xml:78]: remove commented out line

File: pom.xml [hunk 3]
```diff
@@ -75,6 +75,12 @@
         <maven.compiler.release>11</maven.compiler.release>
```
Comment (karianna) [inline pom.xml:75]: Small favor, can you refactor to put properties in alpha order?

File: settings.xml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
```
Comment (karianna) [inline settings.xml:1]: This file should not be included.

File: core/pom.xml [hunk 5]
```diff
@@ -28,6 +28,26 @@
                     </excludePackageNames>
                 </configuration>
             </plugin>
+
+            <plugin>
```
Comment (karianna) [inline core/pom.xml:32]: Is it possible to perhaps put this config in the parent pom and have the children inherit?  Perhaps in a pluginManagement section?

**Review by karianna — CHANGES_REQUESTED** [review]

File: .github/workflows/maven.yml (hunk 1 above)
Comment (dsgrieve) [reply .github/workflows/maven.yml:62]: GITHUB_TOKEN, perhaps?

**Review by dsgrieve — COMMENTED** [review]
Other than the comments from @karianna, I think this looks fine. 

The coverage numbers look horrible! But really it is the parser coverage that matters most.

File: .github/workflows/maven.yml (hunk 1 above)
Comment (anantk24) [reply .github/workflows/maven.yml:62]: Agree with @dsgrieve !

**Comment by karianna** [comment]
@pliakas You'll also need a rebase sorry!

**Comment by dsgrieve** [comment]
This PR is superseded by PR #45

**Pull request closed without merging**